src/osemgrep/cli_scan/Rule_fetching.ml

```diff
@@ -35,7 +35,7 @@ type rules_and_origin = {
   (* TODO? put a config_id: string option? or config prefix? or
    * compute it later based on the origin?
    *)
-  rules : Rule.rules;
+  rules : Rule.lazy_rule list;
   errors : Rule.invalid_rule_error list;
```

Comment:
but if you don't really parse the rule, errors will always be empty?

Reply:
You still can have errors from parsing `id`, `languages`, and `mode`. This is the ugliest thing about my current scheme